hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/metrics/CompressingProbeRenderer.java

```diff
@@ -182,4 +209,14 @@ private void moveNext() {
             }
         };
     }
+
+    private static class MorePublicByteArrayOutputStream extends ByteArrayOutputStream {
```

Comment:
We used this pattern somewhere else I think? Maybe should be a Util class?

Reply:
I did it several times, but it was always removed. This is the only instance now in jet code. And as a rule of thumb, single instance doesn't make it to `Util`. Btw, we should also make it non-synchronized, i reuse the instance and i wouldn't bet the JVM is able to elide locking...